packages/script/src/module.ts

```
@@ -235,6 +235,51 @@ export function applyAutoInject(
+  if (typeof value !== 'string')
+    return
+
+  const trimmed = value.trim()
+  if (!trimmed || (!ABSOLUTE_URL_RE.test(trimmed) && !trimmed.startsWith('//')))
+    return
+
+  try {
+    return new URL(trimmed, 'https://nuxt-scripts.local').hostname || undefined
+  }
+  catch {
+    return
```

ESLint

[error] Unnecessary return statement.

(no-useless-return)